elytron/WFCORE-4172-tls-1.3.adoc

```diff
@@ -0,0 +1,75 @@
+= WFCORE-4172 Add support for TLS 1.3
+:author:            Farah Juma
+:email:             fjuma@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          comma,separated,tags
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+
+JDK 11 has added support for TLS 1.3. This task is to update Elytron to be able to make use of TLS 1.3.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4172[WFCORE-4172]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1022[EAP7-1022]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:okotek@redhat.com[Ondrej Kotek]
+
+=== Affected Projects or Components
+
+* WildFly, Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* It should be possible to configure a ```server-ssl-context``` and ```client-ssl-context``` in the Elytron subsystem
+using the TLS 1.3 protocol and TLS 1.3 https://tools.ietf.org/html/rfc8446#appendix-B.4[cipher suites]. (Note
+that TLS 1.2 cipher suites cannot be used with TLS 1.3 and vice and versa.)
+
+* It should be possible to configure subsystems that reference an ```ssl-context``` to make use of this TLS 1.3 ```server-ssl-context```.
+
+* It should be possible to configure an ```ssl-context``` using the Elytron authentication client using the TLS 1.3
+protocol and TLS 1.3 https://tools.ietf.org/html/rfc8446#appendix-B.4[cipher suites].
+
+* The default cipher suites for TLS 1.3 should be the same as the openssl default cipher suites for TLS 1.3
+(TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_AES_128_GCM_SHA256).
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
```

Comment:
Do I understand it well that this RFE is related just to the Elytron and no implementation for the legacy security will be provided?

Reply:
Correct, this RFE is only related to Elytron.